payroll/views.py

```diff
@@ -141,14 +147,34 @@ def get_context_data(self, **kwargs):
         return context
 
     def median_entity_salary(self):
-        q = Salary.objects.filter(
-            Q(job__position__employer__parent=self.object) |
-            Q(job__position__employer=self.object)
+        entity_qs = Employer.objects.filter(
+            Q(id=self.object.id) | Q(parent_id=self.object.id)
+        )
+
+        median_base_pay = Percentile(
+            "positions__jobs__salaries__amount", 0.5, output_field=FloatField()
+        )
+        median_extra_pay = Percentile(
+            "positions__jobs__salaries__extra_pay", 0.5, output_field=FloatField()
+        )
+        median_total_pay = Percentile(
```

Comment:
There's a subtle issue with this approach that I didn't anticipate. Let's talk it through IRL on Monday morning.

Reply:
The good news: It's fixable with a tiny change.